src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailDistributionDescriptorActionApi.java

```diff
@@ -23,41 +23,91 @@
 import com.synopsys.integration.alert.web.model.TestConfigModel;
-import com.synopsys.integration.exception.IntegrationException;
 
 @Component
-public class EmailDistributionDescriptorActionApi extends DescriptorActionApi {
+public class EmailDistributionDescriptorActionApi extends ChannelDistributionDescriptorActionApi {
     private final EmailGroupChannel emailGroupChannel;
-    private final EmailEventProducer emailEventProducer;
+    private final BlackDuckProjectRepositoryAccessor blackDuckProjectRepositoryAccessor;
 
     @Autowired
     public EmailDistributionDescriptorActionApi(final EmailDistributionTypeConverter databaseContentConverter, final EmailDistributionRepositoryAccessor repositoryAccessor, final EmailGroupChannel emailGroupChannel,
-            final EmailEventProducer emailEventProducer) {
-        super(databaseContentConverter, repositoryAccessor);
+        final BlackDuckProjectRepositoryAccessor blackDuckProjectRepositoryAccessor) {
+        super(databaseContentConverter, repositoryAccessor, emailGroupChannel);
         this.emailGroupChannel = emailGroupChannel;
-        this.emailEventProducer = emailEventProducer;
+        this.blackDuckProjectRepositoryAccessor = blackDuckProjectRepositoryAccessor;
     }
 
     @Override
     public void validateConfig(final Config restModel, final Map<String, String> fieldErrors) {
     }
 
     @Override
-    public void testConfig(final TestConfigModel testConfig) throws IntegrationException {
-        final EmailChannelEvent event = emailEventProducer.createChannelTestEvent((CommonDistributionConfig) testConfig.getRestModel());
-        emailGroupChannel.sendMessage(event);
+    public TestConfigModel createTestConfigModel(final Config config, final String destination) throws AlertFieldException {
+        final EmailDistributionConfig emailDistributionConfig = (EmailDistributionConfig) config;
+
+        final Set<String> emailAddresses = new HashSet<>();
+        Set<BlackDuckProjectEntity> blackDuckProjectEntities = null;
+        if (BooleanUtils.toBoolean(emailDistributionConfig.getFilterByProject())) {
+            blackDuckProjectEntities = blackDuckProjectRepositoryAccessor.readEntities()
+                                           .stream()
+                                           .map(databaseEntity -> (BlackDuckProjectEntity) databaseEntity)
+                                           .filter(databaseEntity -> emailGroupChannel.doesProjectNameMatchThePattern(databaseEntity.getName(), emailDistributionConfig.getProjectNamePattern())
+                                                                         || emailGroupChannel.doesProjectNameMatchAConfiguredProject(databaseEntity.getName(), emailDistributionConfig.getConfiguredProjects()))
+                                           .collect(Collectors.toSet());
+        } else if (emailDistributionConfig.getProviderName().equals(BlackDuckProvider.COMPONENT_NAME)) {
```

Comment:
Put the constant in front of the equals to avoid NPEs: 
BlackDuckProvider.COMPONENT_NAME.equals(emailDistributionConfig.getProviderName())